tests/TESTINS5.MLC

```diff
@@ -341,6 +374,221 @@ BIC_OK2  EQU    *
          IILL   R3,4
          CLG    R3,=AL2(1,2,3,4)
          RT1    CC0
+* MGRK #300
+         LMG   R5,R6,=FL8'2,3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'2,-3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,-3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'123456,10000000'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'1234560000000'
+         RT1    CC0 
+* MS
+         L      R1,=F'2'
+         MS    R1,=F'3'
+         CL     R1,=F'6'
+         RT1  CC0
+         L       R1,=F'2'
+         MS    R1,=F'-3'
+         CL     R1,=F'-6'
+         RT1  CC0
+         L       R1,=F'-2'
+         MS    R1,=F'-3'
+         CL     R1,=F'6'
+         RT1  CC0
+* MSY
+         L      R1,=F'2'
+         MSY    R1,=F'3'
+         CL     R1,=F'6'
+         RT1  CC0
+         L       R1,=F'2'
+         MSY    R1,=F'-3'
+         CL     R1,=F'-6'
+         RT1  CC0
+         L       R1,=F'-2'
+         MSY    R1,=F'-3'
+          CL     R1,=F'6'
+          RT1  CC0
+* MSGRKC
+         LMG   R5,R6,=FL8'2,3'
+         SR    R0,R0
+         MSGRKC  R2,R5,R6
+         RT1    CC2
+         CLG    R2,=FL8'6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'2,-3'
+         MSGRKC  R2,R5,R6
+         CLG    R2,=FL8'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,3'
+         MSGRKC  R2,R5,R6
+         CLG    R2,=FL8'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,-3'
+         MSGRKC  R2,R5,R6
+         CLG   R2,=FL8'6'
+         RT1    CC0 
+         LMG   R5,R6,=XL8'7FFFFFFFFFFFFFFF,2'
+         MSGRKC  R2,R5,R6
+          RT1    CC3  
+* MSRKC
+         LM    R5,R6,=F'2,3'
+         MSRKC  R2,R5,R6
+         CL      R2,=F'6'
+         RT1    CC0 
+         LM     R5,R6,=F'2,-3'
+         MSRKC  R2,R5,R6
+         CL      R2,=F'-6'
+         RT1    CC0 
+         LM     R5,R6,=F'-2,3'
+         MSRKC  R2,R5,R6
+         CL      R2,=F'-6'
+         RT1    CC0 
+         LM     R5,R6,=F'-2,-3'
+         MSRKC  R2,R5,R6
+         CL     R2,=F'6'
+         RT1    CC0 
+         LM   R5,R6,=XL4'7FFFFFFF,2'
+         MSRKC  R2,R5,R6
+         RT1    CC3  
```

Comment:
Same comments as for MSGRKC. Even thouh overflow occurs, check that R2 is set to the expected result.

Also, add the tests for a product of zero.